lua/tardis/libraries/sh_settings.lua

```diff
@@ -80,66 +49,236 @@ function TARDIS:GetSettingsData()
+            TARDIS:GlobalSettingChange(id, value)
+        else
+            error("Setting " .. id .. " is being set clientside, but has unsupported class")
+        end
     end
+
     self:SaveSettings()
-    if (SERVER and broadcast) or networked then
-        self:SendSetting(id,value)
+
+    if (SERVER and data.class == "global") or (CLIENT and data.class == "networked") then
+        self:SendSetting(id, value)
+
+        for k,v in pairs(ents.FindByClass("gmod_tardis")) do
+            v:CallCommonHook("GlobalSettingChanged", id, value)
+        end
     end
+
+    return value
 end
 
-function TARDIS:GetSetting(id,default,ply)
-    if ply and IsValid(ply) and self.ClientSettings[ply:UserID()] and (self.ClientSettings[ply:UserID()][id]~=nil) then
-        return self.ClientSettings[ply:UserID()][id]
-    elseif self.Settings[id] ~= nil then
-        return self.Settings[id]
-    elseif CLIENT and (self.LocalSettings[id] ~= nil) then
-        return self.LocalSettings[id]
-    elseif CLIENT and (self.NetworkedSettings[id]~=nil) then
-        return self.NetworkedSettings[id]
-    else
-        return default
+function TARDIS:GetSetting(id, src)
+    local ply
+    if IsEntity(src) and src:IsPlayer() then
+        ply = src
+    elseif IsEntity(src) then
+        ply = src:GetCreator()
+    end
+
+    if not id then error("Requested setting with no id") end
+    local data = self.SettingsData[id]
+    if not data then error("Requested setting " .. id .. " does not exist") end
+
+    local function select_return_val(table_value)
+        if table_value ~= nil then
+            return table_value
+        end
+        return data.value
+    end
+
+    if data.class == "global" then
+        return select_return_val(self.GlobalSettings[id])
+    end
+
+    if data.class == "local" then
+        if SERVER then
+            error("Local setting " .. id .. "is being requested serverside")
+        end
+        return select_return_val(self.LocalSettings[id])
+    end
+
+    if data.class == "networked" then
+
+        if CLIENT and (ply == nil or ply == LocalPlayer()) then
+            return select_return_val(self.NetworkedSettings[id])
+        end
+
+        if IsValid(ply) then
+            if self.ClientSettings[ply:UserID()] then
+                return select_return_val(self.ClientSettings[ply:UserID()][id])
+            end
+            return select_return_val(nil)
+        end
+
+        error("Networked setting " .. id .. " was requested for invalid player " .. tostring(ply))
     end
+
+    error("Requested setting " .. "either doesn't exist or has no defined class")
+    return
 end
 
-function TARDIS:SendSetting(id,value,ply)
-    net.Start(SERVER and "TARDIS-Settings" or "TARDIS-ClientSettings")
-        net.WriteBool(true)
-        net.WriteType(id)
-        net.WriteType(value)
+--------------------------------------------------------------------------------
+-- Saving
+
+local LOCAL_SETTINGS_FILE = "tardis_settings_cl.txt"
```

Comment:
Are these settings compatible with the previous version or is it a start fresh situation?

Reply:
100% compatible
File names are the same